src/services/upower/device.hpp

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <qcontainerfwd.h>
+#include <qobject.h>
+#include <qqmlintegration.h>
+#include <qtmetamacros.h>
+#include <qtypes.h>
+
+#include "../../core/doc.hpp"
+#include "../../dbus/properties.hpp"
+#include "dbus_device.h"
+
+namespace qs::service::upower {
+
+class UPowerDeviceState: public QObject {
+	Q_OBJECT;
+	QML_ELEMENT;
+	QML_SINGLETON;
+
+public:
+	enum Enum {
+		Unknown = 0,
+		Charging = 1,
+		Discharging = 2,
+		Empty = 3,
+		FullyCharged = 4,
+		PendingCharge = 5,
+		PendingDischarge = 6,
+	};
```

Comment:
Comment PendingCharge and PendingDischarge. Assuming thats when the firmware decides to not charge/discharge until a condition is met but its unclear.